source/fab/util.py

```diff
@@ -48,13 +50,31 @@ def log_or_dot_finish(logger):
 HashedFile = namedtuple("HashedFile", ['fpath', 'file_hash'])
 
 
-def do_checksum(fpath: Path):
+def file_checksum(fpath):
     """
-    Checksum contents of a file.
+    Return a checksum of the given file.
+
+    This function is deterministic, returning the same result across Python invocations.
+
+    We use crc32 for now because it's deterministic, unlike out-the-box hash.
+    We could seed hash with a non-random or look into hashlib, if/when we want to improve this.
```

Comment:
Might consider Adler32 as it is faster than CRC32 and almost as expressive. Or it may not matter.

Reply:
I think it doesn't matter because I've seen Fab take hashes of 3-4k files in <0.1s.